pkg/tfbridge/property_path.go

```diff
@@ -58,6 +63,119 @@ func lookupSchemas(
 	return LookupSchemas(schemaPath, tfs, ps)
 }
 
+func combinePropertyMapKeys(
+	object1, object2 resource.PropertyMap,
+) map[resource.PropertyKey]struct{} {
+	combined := make(map[resource.PropertyKey]struct{})
+	for k := range object1 {
+		combined[k] = struct{}{}
+	}
+	for k := range object2 {
+		combined[k] = struct{}{}
+	}
+	return combined
+}
+
+// SkipChildrenError is an error that can be returned by the visitor to skip the children of the current step.
+type SkipChildrenError struct{}
+
+func (SkipChildrenError) Error() string {
+	return "skip children"
+}
+
+type twoPropertyValueVisitor func(path propertyPath, val1, val2 resource.PropertyValue) error
```

Comment:
The visitor will not be able to distinguish the difference between val1,val2 being null because they are missing form one of the collection, or because they are present in the data as an explicit nil. Perhaps that's ok.

Reply:
I have not encountered a case where the distinction is meaningful. Can revisit if we do.